core/src/main/java/org/apache/calcite/rel/rules/materialize/MaterializedViewRule.java

```diff
@@ -92,6 +95,16 @@
 public abstract class MaterializedViewRule<C extends MaterializedViewRule.Config>
     extends RelRule<C> {
 
+  /**
+   * Weak-key cache of RelMetadataQuery to MaterializationMetadata. The idea is that when
+   * the RelMetadataQuery goes away, so does the cache entry.
+   */
+  private final LoadingCache<RelMetadataQuery, MaterializationMetadata>
+      materializationMetadataCache =
+      CacheBuilder.newBuilder()
```

Comment:
Should we set some parameters?

For example: initialCapacity, maximumSize, refreshAfterWrite, ...

Reply:
> Should we set some parameters?
> 
> For example: initialCapacity, maximumSize, refreshAfterWrite, ...

Through Gc Roots analysis, I found that weakKeys cannot take effect. Even if I delete MaterializationMetadata#RelMetadataQuery, there is still a strong reference pointing to RelMetadataQuery in MaterializationMetadata#RelOptMaterialization#tableRel, so we should really add two parameters, maximumSize and expireAfterAccess, to materializationMetadataCache. Thanks for your suggestion and review!